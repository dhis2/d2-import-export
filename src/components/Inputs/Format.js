import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_JSON = { value: 'json', label: i18n.t('JSON') }
export const OPTION_XML = { value: 'xml', label: i18n.t('XML') }
export const OPTION_PDF = { value: 'pdf', label: i18n.t('PDF') }
export const OPTION_ADX = { value: 'adx', label: i18n.t('ADX') }
export const OPTION_CSV = { value: 'csv', label: i18n.t('CSV') }
export const FORMAT_KEY = 'format'
export const FORMAT_DEFAULT_VALUE = OPTION_JSON.value

export const Format = ({ options }) => (
    <RadioGroup
        name={FORMAT_KEY}
        label={i18n.t('Format')}
        options={options}
        dataTest="input-format"
    />
)

Format.propTypes = {
    options: RadioGroup.propTypes.options,
}
