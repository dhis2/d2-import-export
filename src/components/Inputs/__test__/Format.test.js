import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Format, OPTION_CSV, OPTION_JSON, OPTION_XML } from '../Format'

jest.mock('react-final-form', () => ({
    useField: jest.fn(() => ({
        input: {
            name: 'Name',
            value: '',
            onChange: jest.fn(),
        },
        meta: {
            touched: false,
            error: '',
        },
    })),
}))

describe('Input component - Format', () => {
    it('should render correctly', () => {
        const file = shallow(
            <Format options={[OPTION_JSON, OPTION_XML, OPTION_CSV]} />
        )
        expect(toJson(file)).toMatchSnapshot()
    })
})
