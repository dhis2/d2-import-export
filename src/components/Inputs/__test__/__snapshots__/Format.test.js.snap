// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Format should render correctly 1`] = `
<RadioGroup
  dataTest="input-format"
  label="Format"
  name="format"
  options={
    Array [
      Object {
        "label": "JSON",
        "value": "json",
      },
      Object {
        "label": "XML",
        "value": "xml",
      },
      Object {
        "label": "CSV",
        "value": "csv",
      },
    ]
  }
/>
`;
